completions_test.go

```diff
@@ -1763,13 +1763,15 @@ func TestFlagCompletionWithNotInterspersedArgs(t *testing.T) {
 		Run:       emptyRun,
 		ValidArgs: []string{"arg1", "arg2"},
 	}
-	rootCmd.AddCommand(childCmd, childCmd2)
 	childCmd.Flags().Bool("bool", false, "test bool flag")
 	childCmd.Flags().String("string", "", "test string flag")
 	_ = childCmd.RegisterFlagCompletionFunc("string", func(cmd *Command, args []string, toComplete string) ([]string, ShellCompDirective) {
 		return []string{"myval"}, ShellCompDirectiveDefault
 	})
 
+	// Important: only add the commands after RegisterFlagCompletionFunc was called
+	rootCmd.AddCommand(childCmd, childCmd2)
```

Comment:
I realize we are trying to fix this quickly, so sorry for the nit, but I feel this should be a separate test, and it should mention in a comment issue #1437 to explain why the order is important